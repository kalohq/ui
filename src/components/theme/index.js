/* @flow */
export {default} from './theme';
