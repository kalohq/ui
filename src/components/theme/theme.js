/* @flow */

import tokens from '../../design-tokens/tokens.theme.js';

/**
 * ThemeProvider Kalo Theme
 * 
 * This is the core CSS-in-JS theme that is passed in to the
 * global ThemeProvider. It is mainly composed of our core
 * design tokens.
 */

const theme = {
  space: [0, 8, 16, 32, 64],
  breakpoints: [],
  ...tokens,
};

export default theme;
