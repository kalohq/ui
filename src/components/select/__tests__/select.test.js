/* @flow */
/* eslint-env jest */
import React from 'react';
import {shallow} from 'enzyme';
import renderer from 'react-test-renderer';
import {random} from 'lodash';
import {ThemeProvider} from 'emotion-theming';

import theme from '../../theme';

import Select, {Option} from '../';

const defaultProps = {};

const OPTIONS = ['Hertfordshire', 'Bedfordshire', 'Oxfordshire', 'Hampshire'];

describe('Select', () => {
  const create = props =>
    shallow(
      <ThemeProvider theme={theme}>
        <Select {...defaultProps} {...props}>
          {OPTIONS.map(option => (
            <Option key={option} value={option}>
              {option}
            </Option>
          ))}
        </Select>
      </ThemeProvider>
    );

  test('should render shallow component ok', () => {
    const element = renderer
      .create(
        <ThemeProvider theme={theme}>
          <Select {...defaultProps}>
            {OPTIONS.map(option => (
              <Option key={option} value={option}>
                {option}
              </Option>
            ))}
          </Select>
        </ThemeProvider>
      )
      .toJSON();
    expect(element).toMatchSnapshot();
  });

  test('should render a disabled select', () => {
    const element = renderer
      .create(
        <ThemeProvider theme={theme}>
          <Select {...defaultProps} disabled={true}>
            {OPTIONS.map(option => (
              <Option key={option} value={option}>
                {option}
              </Option>
            ))}
          </Select>
        </ThemeProvider>
      )
      .toJSON();
    expect(element).toMatchSnapshot();
  });

  test('should render a readonly select', () => {
    const element = renderer
      .create(
        <ThemeProvider theme={theme}>
          <Select {...defaultProps} readonly={true}>
            {OPTIONS.map(option => (
              <Option key={option} value={option}>
                {option}
              </Option>
            ))}
          </Select>
        </ThemeProvider>
      )
      .toJSON();
    expect(element).toMatchSnapshot();
  });

  test('should pass all unrecognized props on to the root Block', () => {
    const marginTop = random(10, 20);
    const element = create({marginTop});
    expect(element.props().marginTop).toBe(marginTop);
  });
});
