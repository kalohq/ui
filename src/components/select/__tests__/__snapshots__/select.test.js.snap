// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select should render a disabled select 1`] = `
<div
  aria-disabled={true}
  aria-expanded={false}
  className="ui-select ui-select--disabled"
  data-test="ui-select"
  onSelect={[Function]}
  open={false}
  role="menu"
>
  <div
    className="ui-select__inner"
    onClick={undefined}
  >
    <span
      className="ui-text ui-text--size-small ui-text--weight-normal ui-text--align-unset color-navy500 hover-color-none e1q9f2xp6 css-l4muhg"
      onClick={undefined}
      style={Object {}}
    >
      Select…
    </span>
  </div>
  <div
    className="ui-select__option-group"
    data-test="ui-select__options"
    onClick={[Function]}
  >
    <div
      className="ui-select__option"
      data-test="ui-select__options__option"
      name="Hertfordshire"
      onClick={[Function]}
    >
      <div>
        Hertfordshire
      </div>
    </div>
    <div
      className="ui-select__option"
      data-test="ui-select__options__option"
      name="Bedfordshire"
      onClick={[Function]}
    >
      <div>
        Bedfordshire
      </div>
    </div>
    <div
      className="ui-select__option"
      data-test="ui-select__options__option"
      name="Oxfordshire"
      onClick={[Function]}
    >
      <div>
        Oxfordshire
      </div>
    </div>
    <div
      className="ui-select__option"
      data-test="ui-select__options__option"
      name="Hampshire"
      onClick={[Function]}
    >
      <div>
        Hampshire
      </div>
    </div>
  </div>
</div>
`;

exports[`Select should render a readonly select 1`] = `
<div
  aria-disabled={false}
  aria-expanded={false}
  className="ui-select ui-select--read-only"
  data-test="ui-select"
  onSelect={[Function]}
  open={false}
  role="menu"
>
  <div
    className="ui-select__inner"
    onClick={undefined}
  >
    <span
      className="ui-text ui-text--size-small ui-text--weight-normal ui-text--align-unset color-navy500 hover-color-none e1q9f2xp6 css-l4muhg"
      onClick={undefined}
      style={Object {}}
    >
      Select…
    </span>
  </div>
  <div
    className="ui-select__option-group"
    data-test="ui-select__options"
    onClick={[Function]}
  >
    <div
      className="ui-select__option"
      data-test="ui-select__options__option"
      name="Hertfordshire"
      onClick={[Function]}
    >
      <div>
        Hertfordshire
      </div>
    </div>
    <div
      className="ui-select__option"
      data-test="ui-select__options__option"
      name="Bedfordshire"
      onClick={[Function]}
    >
      <div>
        Bedfordshire
      </div>
    </div>
    <div
      className="ui-select__option"
      data-test="ui-select__options__option"
      name="Oxfordshire"
      onClick={[Function]}
    >
      <div>
        Oxfordshire
      </div>
    </div>
    <div
      className="ui-select__option"
      data-test="ui-select__options__option"
      name="Hampshire"
      onClick={[Function]}
    >
      <div>
        Hampshire
      </div>
    </div>
  </div>
</div>
`;

exports[`Select should render shallow component ok 1`] = `
<div
  aria-disabled={false}
  aria-expanded={false}
  className="ui-select"
  data-test="ui-select"
  onSelect={[Function]}
  open={false}
  role="menu"
>
  <div
    className="ui-select__inner"
    onClick={[Function]}
  >
    <span
      className="ui-text ui-text--size-small ui-text--weight-normal ui-text--align-unset color-navy700 hover-color-none e1q9f2xp6 css-l4muhg"
      onClick={undefined}
      style={Object {}}
    >
      Select…
    </span>
  </div>
  <div
    className="ui-select__option-group"
    data-test="ui-select__options"
    onClick={[Function]}
  >
    <div
      className="ui-select__option"
      data-test="ui-select__options__option"
      name="Hertfordshire"
      onClick={[Function]}
    >
      <div>
        Hertfordshire
      </div>
    </div>
    <div
      className="ui-select__option"
      data-test="ui-select__options__option"
      name="Bedfordshire"
      onClick={[Function]}
    >
      <div>
        Bedfordshire
      </div>
    </div>
    <div
      className="ui-select__option"
      data-test="ui-select__options__option"
      name="Oxfordshire"
      onClick={[Function]}
    >
      <div>
        Oxfordshire
      </div>
    </div>
    <div
      className="ui-select__option"
      data-test="ui-select__options__option"
      name="Hampshire"
      onClick={[Function]}
    >
      <div>
        Hampshire
      </div>
    </div>
  </div>
</div>
`;
